### DISCUSSION TITLE
Handle the 404 error

### DISCUSSION THREAD

**Discussion by ph7jack** [livewire/livewire#3959] [1. Help]
I have a component with a public model.
If this model is deleted in another component, this component throw 404 in the next request
I need to handle the 404 error
The idea is to reset the model property to null when 404 is throw

**Comment by ariaieboy** [comment]
> I have a component with a public model. If this model is deleted in another component, this component throw 404 in the next request I need to handle the 404 error The idea is to reset the model property to null when 404 is throw

you can set an event and when u deleting a record you can call that event and update your component according to that

**Reply by ph7jack** [reply]
events will not work fine. The delete component is separate. The main component broke when call a next request

**Reply by kirkbushell** [reply]
I have this exact same problem - and I've narrowed it down to the hydratePublicProperties middleware. There's a couple of solutions to this, but none are currently available or fixable - it requires, imho - a new method for handling middleware. Effectively, I think it should use a Chain of Command implementation identical to Laravel's. That way, we can actually catch those 404s and handle it nicely.

**Reply by kirkbushell** [reply]
https://github.com/livewire/livewire/discussions/5003

**Comment by jacksontong** [comment]
I have exact same issue, I have 2 components:
DocumentList: render a list of documents, each row has document title, edit/delete button
- when the edit button get pressed, it'll emit an event to `DocumentEditor` to render the document for the user to edit
- when user press the delete button for the active document then I emit another event to `DocumentEditor` to set the document to null, but it throws 404 error instead

```php
<?php
class DocumentList extends Component
{
    public $activeDocument;

    public function deleteDocument($id)
    {
        Document::destroy($id);
        if ($id == $this->activeDocument->id) {
            $this->activeDocument = null;
            $this->emitTo(DocumentEditor::class, 'setDocument', null);
        }
    }
}

// DocumentEditor is not nested inside DocumentList
class DocumentEditor extends Component
{
    public $document;
    protected $listeners = ['setDocument'];
    
    public function setDocument($document)
    {
        $this->document = $document;
    }
}
```
I know that Livewire will try to dehydrate the `$document` on `DocumentEditor` for subsequent requests that's why it thorws 404
but I believe my use case is a common situation.

**Reply by kirkbushell** [reply]
It's very common, also encountered it. It seems this pattern isn't really supported by livewire, as there's no clear solution.

**Reply by kirkbushell** [reply]
I've started a thread here with possible ideas to resolve it:

https://github.com/livewire/livewire/discussions/5003

**Comment by kirkbushell** [comment]
So I figured out a way to manage this, but it's *very* ugly.

I built this piece of middleware:

```php
class CatchLivewire404s
{
    /**
     * @var Request $request
     */
    public function handle($request, \Closure $next)
    {
        if (Str::contains($request->path(), ['livewire/message'])) {
            /** @var Response $response */
            $response = $next($request);

            if ($response->getStatusCode() === 404) {
                info('Model not found during a livewire request.');
                $response->setStatusCode(200);
            }

            return $response;
        }

        return $next($request);
    }
}
```

If you add this as a *global* middleware (necessary because otherwise you can't catch livewire message requests), it'll catch any 404s raised by livewire and instead return a 200.

You can be much more specific if needs be, by simply changing the matched path, or even update it to be a regex or something.
